pkg/cloudprovider/providers/azure/azure_loadbalancer.go

```diff
@@ -455,8 +471,8 @@ func (az *Cloud) cleanupLoadBalancer(clusterName string, service *v1.Service, is
 	return nil
 }
 
-func (az *Cloud) ensurePublicIPExists(serviceName, pipName string) (*network.PublicIPAddress, error) {
-	pip, existsPip, err := az.getPublicIPAddress(pipName)
+func (az *Cloud) ensurePublicIPExists(serviceName, pipResourceGroup string, pipName string) (*network.PublicIPAddress, error) {
+	pip, existsPip, err := az.getPublicIPAddress(pipResourceGroup, pipName)
```

Comment:
I think you might have a problem here if the Public IP in the annotation doesn't exist.  If that happens, won't it attempt to create the Public IP in the cluster resource group?  I'm guessing the desired behaviour would be either to fail or to create it in the resource group from the annotation.  If the latter should it use the ID from the annotation too?  Again the story may be simpler if the annotation specifies only the RG and uses address lookup to locate the PIP resource...

Reply:
Since I'm going to change the annotation to resource group, here I can try to create the pip in the given resource group. If the creation fails, it will return error anyway.